### PR TITLE
feature(stryker): #336 Configurable tmp folder location

### PR DESCRIPTION
**Pull request by brunoqueiros**
Ability to specify a different temporary folder instead of `.stryker-tmp`. Solution based on https://github.com/stryker-mutator/stryker/pull/475

```bash
stryker —tempDir=.another-path-tmp
```

### CODE REVIEW

**Comment by nicojs** [comment]
I've made some changes. 2 things mainly:

* Changed `provideValue` to `provideClass`. With typed-inject's `provideValue('temporaryDirectory', TemporaryDirectory)` you specify to provide the class itself for the `'temporaryDirectory'` token. Using `provideClass` will instruct `typed-inject` to instantiate and cash the temp dir.
* Removed the `instance()` method, since typed-inject will now make sure it is a singleton.

We could still improve on the dependency injection logic. Especially the passing around of the temp dir variable across multiple constructors. But it is already a big improvement.

I didn't update the unit tests or anything, you want to take that? 😅

**Comment by brunoqueiros** [comment]
@nicojs what do you think? ☝️

File: packages/core/test/unit/Stryker.spec.ts [hunk 1]
```diff
@@ -256,10 +256,11 @@ describe(Stryker.name, () => {
         expect(mutationTestExecutorMock.run).calledWith(mutants);
       });
 
-      it('should clean the stryker temp folder', async () => {
+      // TODO: how to test it
+      xit('should clean the stryker temp folder', async () => {
```
Comment (nicojs) [inline packages/core/test/unit/Stryker.spec.ts]: This isn't needed anymore. [`typed-inject`](https://github.com/nicojs/typed-inject) is now responsible for the disposing of it.

**Comment by nicojs** [comment]
@brunoqueiros I've made some changes. Namely:

* Merge master back into this branch
* Document the `tempDirName` option in readme and `StrykerOptions` / `Config`
* Make unit tests pure unit tests instead of testing integration. I.e. use `sinon.createStubInstance(TemporaryDirectory)` instead of `testInjector.injectClass(TemporaryDirectory)` wen not unit testing `TemporaryDirectory` itself

I will merge it if the build is green. 💚

Sorry it took so long. I was on vacation and didn't get around to looking at this until now. 😥

**Pull request merged**